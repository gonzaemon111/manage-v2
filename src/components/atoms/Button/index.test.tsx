import { composeStories } from "@storybook/react";
import { render } from "@testing-library/react";
import * as stories from "./index.stories";
import { Button, LinkButton } from ".";

describe("スナップショットテスト", () => {
  const testCases = Object.values(composeStories(stories)).map((Story) => [
    Story.storyName,
    Story,
  ]);

  test.each(testCases)("Renders %s story", async (storyName, Story) => {
    if (!Story) {
      throw new Error("Story not Found");
    }

    const tree = render(<Story />);
    expect(tree).toMatchSnapshot();
  });
});

describe("Buttonコンポーネント", () => {
  test("適切なclassNameが適用されている", () => {
    const { getByText } = render(<Button name="新規作成" />);
    expect(getByText("新規作成")).toBeTruthy();
  });
});

describe("LinkButtonコンポーネント", () => {
  test("適切なclassNameが適用されている", () => {
    const { getByText } = render(<LinkButton name="新規作成" href="/" />);
    expect(getByText("新規作成")).toBeTruthy();
  });
});
