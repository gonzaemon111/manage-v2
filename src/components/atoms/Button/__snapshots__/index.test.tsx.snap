// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButtonコンポーネント スナップショットテスト Renders Default story 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="text-white bg-primary hover:bg-blue-500 opacity-95 hover:opacity-90 transition-all duration-200 transform focus:ring-4 focus:outline-none font-medium text-sm px-10 py-2.5 text-center shadow-md hover:shadow-xl inline-flex items-center gap-2 mr-2 mb-2"
        type="button"
      >
        新規作成
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text-white bg-primary hover:bg-blue-500 opacity-95 hover:opacity-90 transition-all duration-200 transform focus:ring-4 focus:outline-none font-medium text-sm px-10 py-2.5 text-center shadow-md hover:shadow-xl inline-flex items-center gap-2 mr-2 mb-2"
      type="button"
    >
      新規作成
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`スナップショットテスト Renders Default story 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="text-white bg-primary hover:bg-blue-500 opacity-95 hover:opacity-90 transition-all duration-200 transform focus:ring-4 focus:outline-none font-medium text-sm px-10 py-2.5 text-center shadow-md hover:shadow-xl inline-flex items-center gap-2 mr-2 mb-2"
        type="button"
      >
        新規作成
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text-white bg-primary hover:bg-blue-500 opacity-95 hover:opacity-90 transition-all duration-200 transform focus:ring-4 focus:outline-none font-medium text-sm px-10 py-2.5 text-center shadow-md hover:shadow-xl inline-flex items-center gap-2 mr-2 mb-2"
      type="button"
    >
      新規作成
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
